spring-cloud-generator/pom.xml

```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>spring-cloud-gcp</artifactId>
+        <groupId>com.google.cloud</groupId>
+        <version>4.1.5-SNAPSHOT</version><!-- {x-version-update:spring-cloud-gcp:current} -->
+    </parent>
+
+    <groupId>com.google.api</groupId>
+    <artifactId>spring-cloud-generator</artifactId>
+    <name>Spring Code Generator for Google Cloud</name>
+    <description>Spring Code Generator for Google Cloud</description>
+
+    <properties>
+        <!-- [gapic-test]: for compiling test protos -->
+        <protobuf.version>3.21.12</protobuf.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.google.api</groupId>
+            <artifactId>gapic-generator-java</artifactId>
+            <version>2.17.0</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.google.cloud</groupId>
+            <artifactId>spring-cloud-gcp-autoconfigure</artifactId>
+        </dependency>
+        <dependency>
+            <!-- [gapic-test]: dependency from using gapic-generator java testing framework-->
+            <groupId>io.github.java-diff-utils</groupId>
+            <artifactId>java-diff-utils</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <extensions>
+            <extension>
+                <!-- [gapic-test]: for use in protobuf-maven-plugin below -->
+                <groupId>kr.motd.maven</groupId>
+                <artifactId>os-maven-plugin</artifactId>
+                <version>1.7.1</version>
+            </extension>
+        </extensions>
+        <plugins>
+            <plugin>
+                <!-- [gapic-test]: for compiling test protos -->
```

Comment:
Is this an attempt to replace a Bazel target with a Maven plugin? If so, what's the benefit over keeping it in Bazel?

Reply:
I think this is more to make the generation process similar to gapic-generator-java, meaning that we build a `spring-cloud-generator` jar with Maven first and use it in the generation process. I don't think we can run tests through Maven without this change, similar to what [gapic-generator-java](https://github.com/googleapis/gapic-generator-java/blob/eee46fe2cbc466549115b943a79d7863bdc74c40/gapic-generator-java/pom.xml#L248) is doing.

Reply:
Ah yeah this is part of copying over testing setup from gapic-generator-java - specifically it's for compiling protos used for unit tests ([source](https://github.com/googleapis/gapic-generator-java/blob/eee46fe2cbc466549115b943a79d7863bdc74c40/gapic-generator-java/pom.xml#L248)) (allowing `TestProtoLoader` to parse the proto).   

It can later be removed given a published test jar from gapic-generator-java as discussed; currently this setup allows `mvn test -DupdateUnitGoldens` to work in this submodule.